src/effects/VST/VSTEffect.cpp

```diff
@@ -1258,6 +1260,49 @@ bool VSTEffect::LoadSettings(
    return true;
 }
 
+
+bool VSTEffect::SaveSettings(const EffectSettings& settings, CommandParameters& parms) const
+{
+   const VSTEffectSettings& vstSettings = GetSettings(settings);
+
+   for (const auto& item : vstSettings.mParamsMap)
+   {
+      const wxString& name  = item.first;
+      const float&    value = item.second;
+
+      if (!parms.Write(name, value))
+      {
+         return false;
+      }
+   }
+
+   return true;
+}
+
+bool VSTEffect::LoadSettings(const CommandParameters& parms, EffectSettings& settings) const
+{
+   VSTEffectSettings& vstSettings = GetSettings(settings);
+   vstSettings.mParamsMap.clear();
+
+   long index{};
+   wxString key;
+   float value = 0.0f;
+   if (parms.GetFirstEntry(key, index))
+   {
+      do
+      {
+         if (parms.Read(key, value))
+            vstSettings.mParamsMap[key] = value;
+         else
+            return false;
+
+      } while (parms.GetNextEntry(key, index));
+   }
```

Comment:
This new definition of `LoadSettings` should set up all members of `vstSettings` from config file information, not just the parameter map.

Reply:
This is what I added now, I hope it is what you meant?

 ```
   vstSettings.mChunk     = std::nullopt;
   vstSettings.mVersion   = VSTEffectWrapper::mVstVersion;
   vstSettings.mUniqueID  = VSTEffectWrapper::mAEffect->uniqueID;
   vstSettings.mNumParams = VSTEffectWrapper::mAEffect->numParams;
```

Reply:
Yes